deepspeed/checkpoint/deepspeed_checkpoint.py

```diff
@@ -270,12 +276,17 @@ def _merge_state_dicts(self, sd_list):
 
         return merged_sd
 
-    def _validate_folder(self, dir):
+    def _validate_folder(self, dir,pipeline_parallel=False):
         basic_folder_validation(dir)
 
         file_list = get_files(dir)
```

Comment:
I think we can examine `file_list` to determine if `layer*` files exist and thus distinguish PP and non-PP runs. In general, I don't like the idea of introducing `--pipeline-parallel`.